cookbook/security/guard-authentication.rst

```diff
@@ -0,0 +1,577 @@
+            </service>
+        </services>
+
+    .. code-block:: php
+
+        // app/config/services.php
+        use Symfony\Component\DependencyInjection\Definition;
+        use Symfony\Component\DependencyInjection\Reference;
+
+        $container->setDefinition('app.token_authenticator', new Definition(
+            'AppBundle\Security\TokenAuthenticator',
+            array(new Reference('doctrine.orm.entity_manager'))
+        ));
+
+Finally, configure your ``firewalls`` key in ``security.yml`` to use this authenticator:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/security.yml
+        security:
+            # ...
+
+            firewalls:
+                # ...
+
+                main:
+                    anonymous: ~
+                    logout: ~
+
+                    guard:
+                        authenticators:
+                            - app.token_authenticator
+
+                    # if you want, disable storing the user in the session
+                    # stateless: true
+                    
+                    # maybe other things, like form_login, remember_me, etc
+                    # ...
+
+    .. code-block:: xml
+
+        <!-- app/config/security.xml -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <srv:container xmlns="http://symfony.com/schema/dic/security"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:srv="http://symfony.com/schema/dic/services"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+            <config>
+                <!-- ... -->
+
+                <firewall name="main"
+                    pattern="^/"
+                    anonymous="true"
+                >
+                    <logout />
+
+                    <guard>
+                        <authenticator>app.token_authenticator</authenticator>
+                    </guard>
+
+                    <!-- ... -->
+                </firewall>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        // app/config/security.php
+
+        // ..
+
+        $container->loadFromExtension('security', array(
+            'firewalls' => array(
+                'main'       => array(
+                    'pattern'        => '^/',
+                    'anonymous'      => true,
+                    'logout'         => true,
+                    'guard'          => array(
+                        'authenticators'  => array(
+                            'app.token_authenticator'
+                        ),
+                    ),
+                    // ...
+                ),
+            ),
+        ));
+
+You did it! You now have a fully-working API token authentication system. If you're
+homepage required ``ROLE_USER``, then you could test it under different conditions:
```

Comment:
It's a conditional sentence using the subjunctive tense (I think): language minutiae, but I believe `required` is correct (but really, both are fine)